### ISSUE TITLE
@ApiImplicitParameter 'dataType' isn't detected.

### ISSUE THREAD

**Issue by dvisa** [springfox/springfox#1141]
Hi,

I'm unable to generate/see the public static Request class in Swagger2(springfox v2.3.1) documentation when it is specified as @ApiImplicitParam annotation and not referenced globally in any class barring some local variables in handler method. It is simply not detected by Springfox as model or argument.

If I add an member of type Request in "Response" without ever assigning it a value. The documentation gets generated!

Any hints? Does this need some fixes in springfox?

Thanks,
Deep 
```java
@Controller 
public class RequestController {
// ...
@ApiOperation(value = "test", position = 2, nickname = "test", notes = "Implementation notes ")
@ApiImplicitParams({@ApiImplicitParam(dataType = "Request", name = "request", paramType = "body", required=true)})
@ApiResponses(value = {@ApiResponse(code = 200, message = "message.",response = Response.class),})
@RequestMapping(value = "/test", method = RequestMethod.POST, produces = "application/json")
public void handle(@ApiIgnore @RequestBody String requestJson, HttpServletRequest request, HttpServletResponse response) throws IOException {
    // Using Object Mapper local instance to retrieve RequestJson:
    ObjectMapper mapper = new ObjectMapper();
    Request req;
    try {
      req = mapper.readValue(requestJson, Request.class);
    } catch (Exception ex) {
      // ...
    }
    // Request processing
}
    public static class Request {
        public int id;
        //... parsed from json by Object mapper.
    }
    public static class Response {
        public VariousJson object1;
        //...
        // If I uncomment following here, it works but it's ugly.
        // Request     dummyUnusedReq;
    }
//...
}
```

**Comment by dilipkrish** [comment]
Have you looked [at this documentation](http://springfox.github.io/springfox/docs/current/#overriding-property-datatypes) regarding overriding data types?

**Comment by dvisa** [comment]
Yes, read that and I tried to use the qualified class name by referring to the Request in data type as "dataType="packageName.RequestController.Request", it didn't help.

I also did couple of other experiments such as moving the class outside "RequestController". It is still not detected.

It works only if I add "Request" parameter as a member of some existing/detected model class.

**Comment by dilipkrish** [comment]
@dvisa This is a duplicate of #735 

The work around for this is to add a model like you have done or the on suggested in #735 (which is less desirable since the model you're adding is an input model). This should be fixed in 2.4 where we'll allow you to document additional models not inferred via controller methods.
